src/VisualStudio/VisualStudioDiagnosticsToolWindow/Remote/RemoteHostClientFactory.cs

```diff
@@ -0,0 +1,27 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Composition;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.VisualStudio.LanguageServices.Remote;
+using Roslyn.VisualStudio.Test.Utilities.Remote;
+
+namespace Roslyn.VisualStudio.DiagnosticsWindow.Remote
+{
+    [ExportWorkspaceService(typeof(IRemoteHostClientFactory), layer: ServiceLayer.Host), Shared]
```

Comment:
our workspace service has a way to choose one from others when it exports same service. I made one in diagnostic windows vsix to be picked over one in VS.Next.

when option is off, it will return one in VS.Next otherwise, one in diagnostic windows vsix.